src/squlearn/kernel/matrix/fidelity_kernel.py

```diff
@@ -101,8 +101,7 @@ def __init__(
         self._enc_circ = self._encoding_circuit.get_circuit(
             self._feature_vector, self._parameter_vector
         )
-
-        if "statevector_simulator" in str(self._executor._backend):
+        if "statevector" in self._executor._backend.configuration().backend_name:
```

Comment:
is_statevector is missing here

Reply:
You're right, thanks for the pointer 😄 
```suggestion
        if self._executor.is_statevector:
```